doc/source/configuration/kayobe-automation.rst

```diff
@@ -0,0 +1,125 @@
+Kayobe automation provides users an easy process of performing tasks such as; overcloud service deploy, config-diff, tempest testing and many more.
+With it being integrated into platforms such as GitHub or GitLab it builds a close relationship between the contents of the deployments kayobe configuration and what is currently deployed.
+This is because operations such as opening a pull request will trigger a config diff to be generated providing insight on what impact it might have on services or a tempest test that could be scheduled to run daily providing knowledge of faults earlier than before.
+
+Kayobe automation has been designed to be independent of any CI/CD platform with all tasks running inside of a purpose build Kayobe container.
+However, platform specific workflows need to be deployed to bridge the gap between the contents of Kayobe Automation and these CI/CD platforms.
+To achieve this work has been carried to template workflows as deployment specific choices have to be made when writing workflows, such as; multiple environment support, container registry location and secret sharing.
+The templating of workflows is offered through the `stackhpc.kayobe_workflows <https://github.com/stackhpc/ansible-collection-kayobe-workflows/>`__ collection which currently supports GitHub workflows and should enable the easy and error free deployment of workflows.
+
+GitHub Deployment
+=================
+
+To enable Kayobe Automation where GitHub Actions is used please follow the steps described below starting with the deployment the runners.
+
+Runner Deployment
+-----------------
+
+1. Identify a suitable host for hosting the runners.
+    GitHub runners need to be deployed on a host which has not had Docker deployed using kolla.
+    This is because GitHub runners cannot provide `network options when running in a container <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontaineroptions>`__.
+
+    Ideally an Infra VM could be used here or failing that the control host.
+    Wherever it is deployed the host will need access to the :code:`admin_network`, :code:`public_network` and the :code:`pulp registry` on the seed.
+
+2. Edit the environments :code:`inventory/groups` to add the predefined :code:`github-runners` group to :code:`infra-vms`
+
+.. code-block:: ini
+
+    [infra-vms:children]
+    github-runners
+
+3. Edit the environments :code:`inventory/hosts` to define the host(s) that will host the runners.
+
+.. code-block:: ini
+
+    [github-runner]
+    runner-01
+
+4. Provide all the relevant Kayobe :code:`group_vars` for :code:`github-runners` under :code:`inventory/hosts/github-runners`
+    * `infra-vms` ensuring all required `infra_vm_extra_network_interface` are defined
+    * `network-interfaces`
+    * `python-interpreter.yml` ensuring that `ansible_python_interpreter: /usr/bin/python3` has been set
```

Comment:
Wouldn't this be image-dependent?

Reply:
I don't believe so. The reason for this to be defined would be because the default python interpreter might not be present or the user `runner_user` lacks permission to use.

Reply:
Does the normal method of using the Kayobe virtualenv not work?